### PR TITLE
Issue #12319: validate PR number when commit is for PR

### LINKED ISSUE #12319: CI: validate PR number when commit is for PR
Identified at https://github.com/checkstyle/checkstyle/pull/12313#issuecomment-1287241561 ,

As title suggests, when a commit message references `Pull #`, it should validate the number to be the same number of the current PR.

We have never references other PRs with this commit message, and we shouldn't since release notes will be pulling in the title for the PR number given to it.

### PR DESCRIPTION
**Pull request by nrmancuso**
Closes #12319 

~~First commit is not under review.~~

Example of failure (on commit https://github.com/checkstyle/checkstyle/pull/12344/commits/0bf87981c766048276a61e9d1e5f9458f0781ec4): https://app.circleci.com/pipelines/github/checkstyle/checkstyle/15747/workflows/45589939-13ca-4595-b983-796e6cb95b01/jobs/190021

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -248,6 +248,27 @@ jobs:
...
+  check-pull-number:
+    docker:
+      - image: cimg/base:2022.10
+    steps:
+      - checkout
+      - run:
+          name: Check that 'Pull#XXXX' matches PR number
+          command: |
+            COMMITS="$(git log --format=format:%B master.."$CIRCLE_SHA1")"
+
```
Comment (romani) [inline .circleci/config.yml:260]: Move it validation.sh or similar sh file

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: .circleci/config.yml (hunk 1 above)
Comment (nrmancuso) [reply .circleci/config.yml]: done

File: .circleci/config.yml [hunk 2]
```diff
@@ -229,6 +229,29 @@ jobs:
...
+  check-pull-number:
+    docker:
+      - image: cimg/base:2022.10
+    steps:
+      - checkout
+      - run:
+          name: Check that 'Pull#XXXX' matches PR number
+          command: |
+            export PR_HEAD_SHA=$CIRCLE_SHA1
+            export PR_NUMBER=$CIRCLE_PR_NUMBER
```
Comment (nrmancuso) [inline .circleci/config.yml:252]: reassigned these to make script CI agnostic

**Comment by romani** [comment]
GitHub, rebase

**Review by romani — APPROVED** [review]

**Pull request merged**